app/connectors/SubscriptionConnector.scala

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package connectors
+
+import java.time.ZonedDateTime
+import java.time.format.DateTimeFormatter
+import java.util.UUID
+
+import config.AppConfig
+import javax.inject.Inject
+import models.DisplaySubscriptionForDACRequest
+import uk.gov.hmrc.http.logging.Authorization
+import uk.gov.hmrc.http.{HeaderCarrier, HttpClient, HttpResponse}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class SubscriptionConnector @Inject()(val config: AppConfig, val http: HttpClient) {
+
+  def displaySubscriptionForDAC(subscriptionForDACRequest: DisplaySubscriptionForDACRequest)
+                               (implicit hc: HeaderCarrier, ec: ExecutionContext): Future[HttpResponse] = {
+
+    val displaySubscriptionUrl = s"${config.registrationUrl}/dac6/dct04/v1"
+    val conversationID = subscriptionForDACRequest.displaySubscriptionForDACRequest.requestCommon.conversationID.getOrElse("")
```

Comment:
If we can't retrieve it, EIS will throw a 400 because it requires the x-conversation-id to match the conversationID. Like you said, this shouldn't happen so it should never go to the else.

Reply:
Ok as long as the error is thrown and logged somewhere I will approve